sunpy/timeseries/timeseriesbase.py

```diff
@@ -445,6 +445,33 @@ def concatenate(self, others, same_source=False, **kwargs):
         object._sanitize_units()
         return object
 
+    def resample(self, rule, **kwargs):
+        """
+        Returns a resampled time series.
+
+        Parameters
+        ----------
+        rule : `~pandas.tseries.offsets.DateOffset`, `~pandas.Timedelta` or `str`
+            The offset string or object representing target conversion.
+
+        Returns
+        -------
+        `~sunpy.timeseries.TimeSeries`
+            A resampled `~sunpy.timeseries.TimeSeries`.
+
+        Notes
+        -----
+        This method is a wrapper around `~pandas.DataFrame.resample`; all additional
+        keyword arguments are passed to this method.
+        """
+
+        object = GenericTimeSeries(self._data.resample(rule, **kwargs).pad(),
```

Comment:
What does the pad do in this context?

Reply:
It'll fill the missing values with `NaN`